lib/common.sh

```diff
@@ -104,15 +104,16 @@ clone_repo() {
   if [[ -d "${REPO_PATH}" ]] && [[ "${FORCE_REPO_UPDATE}" = "true" ]]; then
     rm -rf "${REPO_PATH}"
   fi
-  if [[ ! -d "${REPO_PATH}" ]] ; then
+  if [[ ! -d "${REPO_PATH}" ]]; then
     pushd "${M3PATH}" || exit
-    git clone "${REPO_URL}" "${REPO_PATH}"
-    popd || exit
-    pushd "${REPO_PATH}" || exit
-    git checkout "${REPO_BRANCH}"
-    git checkout "${REPO_COMMIT}"
-    git pull -r || true
+    git clone --depth 1 --branch "${REPO_BRANCH}" "${REPO_URL}" "${REPO_PATH}"
```

Comment:
I worry this may cause us pain down the road. How can we make sure that cloning with depth 1 will be sufficient for every repo in the future?

IMO the way to improve speed would be to reduce the needs of cloning the repos in the first place.

Reply:
In CI environments depth 1 should be enough.
If a user runs dev-env as a local sandbox then it is possible to substitute the automatically cloned repos with manually cloned repos thus if a user needs additional history it can be provided already.  

But please correct me if I have missed something. I can't think of a use-case at the moment that we can't handle with this structure.